app/uk/gov/hmrc/tai/repositories/EmploymentRepository.scala

```diff
@@ -37,168 +35,125 @@ class EmploymentRepository @Inject()(
   rtiConnector: RtiConnector,
   cacheConnector: CacheConnector,
   npsConnector: NpsConnector,
-  auditor: Auditor) {
+  employmentBuilder: EmploymentBuilder) {
 
   private val EmploymentMongoKey = "EmploymentData"
 
   def employment(nino: Nino, id: Int)(
-    implicit hc: HeaderCarrier): Future[Either[EmploymentRetrievalError, Employment]] =
-    employmentsForYear(nino, TaxYear()) flatMap { empForYear =>
-      if (empForYear.exists(_.annualAccounts.exists(_.realTimeStatus == TemporarilyUnavailable))) {
-        Future.successful(Left(EmploymentAccountStubbed))
+    implicit hc: HeaderCarrier): Future[Either[EmploymentRetrievalError, Employment]] = {
+    val taxYear = TaxYear()
+    employmentsForYear(nino, taxYear) map { empForYear =>
+      if (empForYear.exists(_.tempUnavailableStubExistsForYear(taxYear))) {
+        Left(EmploymentAccountStubbed)
       } else {
-        fetchEmploymentFromCache(nino) map { emp =>
-          emp.find(_.sequenceNumber == id) match {
-            case Some(employment) => Right(employment)
-            case None => {
-              val sequenceNumbers = emp.map(_.sequenceNumber).mkString(", ")
-              Logger.warn(s"employment id: $id not found in employment sequence numbers: $sequenceNumbers")
-              Left(EmploymentNotFound)
-            }
+        empForYear.find(_.sequenceNumber == id) match {
+          case Some(employment) => Right(employment)
+          case None => {
+            val sequenceNumbers = empForYear.map(_.sequenceNumber).mkString(", ")
+            Logger.warn(s"employment id: $id not found in employment sequence numbers: $sequenceNumbers")
+            Left(EmploymentNotFound)
           }
         }
       }
     }
+  }
 
-  def employmentsForYear(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[Employment]] =
-    fetchEmploymentFromCache(nino).flatMap { allEmployments =>
-      allEmployments
-        .filter(_.annualAccounts.exists(_.taxYear == year))
-        .map(e => e.copy(annualAccounts = e.annualAccounts.filter(_.taxYear == year))) match {
-        case Nil                     => employmentsFromHod(nino, year)
-        case employmentsForGivenYear => Future.successful(employmentsForGivenYear)
-      }
+  def employmentsForYear(nino: Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier): Future[Seq[Employment]] =
```

Comment:
If this returned a Employments object it might help simplify the calling code to this function

Reply:
as above